conftest.py

```diff
@@ -62,6 +62,27 @@ def skip_warnings(pytestconfig):
     return skip_warnings
 
 
+@pytest.fixture(scope="session")
+def require_warnings_skipped(skip_warnings):
+    """Require '--skipwarnings' CLI option."""
+    # skip_warnings=True implies that pytest is being run with warning
+    # reporting ENABLED. So, proceed to warning test.
+    # skip_warnings=False implies that pytest is being run with
+    # -p no:warnings, and thus the below warning will be suppressed.
+    # Thus, for the latter case, a forced fail is appropriate
+    assert skip_warnings
+
+
+@pytest.fixture
+def enable_warnings_plugin(item, config, require_warnings_skipped):
```

Comment:
I do like this refactor and separation of responsibility. Dunno if we'll specifically need `require_warnings_skipped` on its own, but can't hurt to have it isolated.